### ISSUE TITLE
possibility to show map on train.txt valid.txt

### ISSUE THREAD

**Issue by slinshady89** [AlexeyAB/darknet#2838]
As I understood correctly the map uses the validation data set and calculates the map on these.
`./darknet detector map data/obj.data yolo-obj.cfg backup\yolo-obj_7000.weights`
Is there an option to visualize the map on the training and the validation data separatly in the graph so if an increasing gap is seen I could break and train with another learning rate or change the model itself to pretend overfitting earlier in the training process?

**Comment by drapado** [comment]
You can't do it in the graph while training. However, what you can do is run the command: `./darknet detector map ...` while having in the obj.data file valid=valid.txt; write down the results, then change in the obj.data file valid=train.txt and run `./darknet detector map ...`. It will show you know the map on the train set

**Comment by slinshady89** [comment]
yeah I thought of that too but that way I have to babysit much more and stop training to check for the map . I already tried to find where the map is calculated and drawn and just do the same routine afterwards for the train.txt in another color but can't find it. 
I'm actually watching this cool course of stanford university which brought me to that. check 
 Train/Val accuracy in [http://cs231n.github.io/neural-networks-3/](url)

**Comment by alemelis** [comment]
calculating the mAP on the training set may be cumbersome as that set should be way larger that the validation one

imho, If you really want to do that check, it may be easier to monitor avg_loss and validation mAP and only check the test mAP once you see the two above plateauing

otherwise, just duplicate the code for map calculation pointing to the train.txt instead of valid.txt

**Comment by AlexeyAB** [comment]
> visualize the map on the training and the validation data separatly in the graph so if an increasing gap is seen I could break and train with another learning rate or change the model itself to pretend overfitting earlier in the training process?

Why do you want to see the gap instead of validation mAP plateau?

To get the maximum mAP as quickly as possible, you usually have to train while the mAP increases, and only after the plateau should you reduce your learning rate.

**Comment by slinshady89** [comment]
ahhh okay. so that's maybe not necessary at all. Thought it's maybe useful but on the other side you probably get a better feeling if your more into that like me as a beginner :)
